### PR TITLE
Security audit-inspired fixes

### PR DESCRIPTION
**Pull request by philipcmonk**
I'm going to keep adding fixes here, one per commit.  It needn't be merged until they've all been added.

### CODE REVIEW

**Comment by philipcmonk** [comment]
a2402b7 has some subtlety to it that I'd like you to pay particular attention to.  Do refer to sections 6.5 and 6.6 of the security audit to see the motivation, and then make sure you understand why it's okay to cast the address to `NumeraireBackend`.

**Comment by philipcmonk** [comment]
4f5796b fixes at least two potential overflow bugs, which I point out in line comments on the commit here on Github.

I've completed this round of fixes, and this is ready to be merged once it's reviewed.

File: contracts/NumeraireShared.sol [hunk 1]
```diff
@@ -56,13 +56,12 @@ contract NumeraireShared is Safe {
...
-        uint256 total_allowance = initial_disbursement + incremental_allowance;
-        assert(safeToSubtract(total_allowance, total_minted));
-        return total_allowance - total_minted;
+        return
+            safeSubtract(
+                safeAdd(initial_disbursement,
+                    safeMultiply(weekly_disbursement,
+                        safeSubtract(block.timestamp, deploy_time))
+                    / 1 weeks),
+                total_minted);
```
Comment (philipcmonk) [inline contracts/NumeraireShared.sol:65]: Pay particular attention to this.  I believe it's correct, but it's the most significant change in this commit.

Comment (xanderdunn) [reply contracts/NumeraireShared.sol:65]: Looks good

**Review by xanderdunn — APPROVED** [review]

**Pull request merged**